Feature sb 22 add sign up fe

ValentinaTenus opened the pull request:
1. created reusable components (BaseButton with custom variants and CustomFormGroup)
2. auth slice and auth api
3. baseQuery and baseQueryWithReauth for managing queries with token and without
4. auth page 
5. sign-up form  with useSignUpForm hook for logic
6. validation according to requeres
7. added fonts, color, button and typography variants to theme
8. added some changes in eslint, typescript configurations(for instanse, run eslint before commit)
![sign-up-form-small](https://github.com/ZenBit-Tech/sunny-bunnies_fe/assets/120641395/dd2bd854-271a-4ebd-8318-7c245e0c8300)

![sign-up-page](https://github.com/ZenBit-Tech/sunny-bunnies_fe/assets/120641395/5fe4454e-210a-44ce-834d-1ac23312fccc)
![Uploading sign-up-with-errors.png…]()


*  bottom links aligned

lyudad commented on src/components/auth/google-button/google-button.tsx:
```diff
@@ -0,0 +1,30 @@
+			fullWidth
+			sx={{
+				"&:hover": { backgroundColor: "#EDEAE9" },
+				backgroundColor: "#EDEAE9",
```
1.3.2

ValentinaTenus replied:
moved to constants

lyudad commented on src/components/auth/sign-in-form/sign-in-form.tsx:
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const SignInForm: React.FC = () => {
+	return <>Log in</>;
```
1.3.3

ValentinaTenus replied:
done

lyudad commented on src/components/auth/sign-up-form/sign-up-form.tsx:
```diff
@@ -0,0 +1,98 @@
+					<CustomFormGroup
+						control={control}
+						error={errors.password}
+						label="Password*"
```
1.3.3

ValentinaTenus replied:
done

lyudad commented on src/components/common/base-button/base-button.tsx:
rename to index.tsx

ValentinaTenus replied:
done

lyudad commented on src/libs/helpers/storage.ts:
```diff
@@ -0,0 +1,37 @@
+const storageKey = {
```
I assume redux-persist library will replace this

ValentinaTenus replied:
replaced storage with redux-persist

ValentinaTenus commented:
1. I've fixed your remarks
2. added private and public routes
3. saved user data and tokens after sign up(we will change it after adding email verification)

lyudad commented on src/components/auth/auth-links/auth-links.tsx:
```diff
@@ -0,0 +1,56 @@
+			</Typography>
+		</Link>
+	);
+};
```
there are 2 components in one file, never do it

ValentinaTenus replied:
done
